plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -113,7 +113,19 @@ public Object buildAll(String modelName, Project rootProject) {
         gradleSourceSet.setCompilerArgs(getCompilerArgs(project, sourceSet));
         gradleSourceSets.add(gradleSourceSet);
       });
+    }
 
+    // run through twice to cope with any name clashes
+    for (DefaultGradleSourceSet sourceSet : sourceSetMap.values()) {
+      if ("main".equals(sourceSet.getSourceSetName())) {
+        sourceSet.setDisplayName(stripPathPrefix(sourceSet.getProjectPath()));
+      }
+    }
+    for (DefaultGradleSourceSet sourceSet : sourceSetMap.values()) {
+      if (!"main".equals(sourceSet.getSourceSetName())) {
+        String uniqueName = createUniqueDisplayName(sourceSet, sourceSetMap);
```

Comment:
@Arthurm1 I just had this new question in my mind when I was trying to refactor the code.

Why not just directly append the sourceset name to the project path string? for example:

`:a:b [main]` & `:a:b [test]`.

By doing this, we can remove the logic to check duplicated names.

Reply:
You can.  But I was trying to get the name as simple as possible for the user.

If you look at Gradle tasks for the `model` project you have tasks like...
main source set: `classes, compileJava`, test source set: `testClasses, compileTestJava`

So Gradle tries to just ignore the `main` in the `main` source set.

Also people use text based editors e.g. vim, emacs, helix etc. and CLI for BSP so it's good to have these names as simple and short as possible if you ever have to type them.
I'd like to avoid special characters (space, braces) for this same reason.  But that might be more an issue for the client to deal with than the server as there's nothing in the BSP spec about displayName limitations.

I guess it's personal preference.
This...
![image](https://github.com/microsoft/build-server-for-gradle/assets/5612295/1762b712-9824-4ccf-bfdf-aa922aa2fc31)
Or this...
![image](https://github.com/microsoft/build-server-for-gradle/assets/5612295/4547ba9d-a345-476a-87fa-819d7c575aa0)

I find the first one easier on the eye.

Reply:
I see. Thank you for the explanation. Makes sense since Gradle has similar concept for the tasks.

> I guess it's personal preference.

Yes. Because I feel that when there is a conflict, things like `foo-test` & `foo-test2`might be a little bit strange. It's just my personal opinion, and I don't have a strong opinion which one is better.